src/ClusterBootstrap/az_tools.py

```diff
@@ -921,6 +921,10 @@ def gen_cluster_config(output_file_name, output_file=True, no_az=False):
 
     if "sku_mapping" in config:
         cc["sku_mapping"] = config["sku_mapping"]
+        for sku in cc["worker_sku_cnt"]:
+            # this means that the cluster deployed with this pipeline cannot be heterogeneous
+            cc["gpu_type"] = cc["sku_mapping"].get(sku, {}).get('gpu-type', "None")
+            break
```

Comment:
Why is here a `break`?

Reply:
because old pipeline is based on the assumption that the cluster is homogeneous. so there could be only one gpu_type